nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/EnforceOrder.java

```diff
@@ -0,0 +1,538 @@
+            maxOrderProperty = processContext.getProperty(MAX_ORDER);
+        }
+
+        private void setFlowFile(final FlowFile flowFile) {
+            this.flowFile = flowFile;
+            this.groupId = null;
+            this.order = null;
+        }
+
+        private boolean computeGroupId() {
+            groupId = groupIdentifierProperty.evaluateAttributeExpressions(flowFile).getValue();
+            if (isBlank(groupId)) {
+                transferToFailure(flowFile, "Failed to get Group Identifier.");
+                return false;
+            }
+            return true;
+        }
+
+        private boolean computeOrder() {
+            try {
+                order = getOrder.apply(flowFile);
+            } catch (final NumberFormatException e) {
+                transferToFailure(flowFile, "Failed to parse order attribute due to " + e, e);
+                return false;
+            }
+            return true;
+        }
+
+        private boolean computeMaxOrder() {
+            if (maxOrderProperty.isSet()) {
+                // Compute maxOrder for this group if it's not there yet.
+                final String maxOrderStr = groupStates.computeIfAbsent(STATE_MAX_ORDER.apply(groupId),
+                        k -> maxOrderProperty.evaluateAttributeExpressions(flowFile).getValue());
+                if (isBlank(maxOrderStr)) {
+                    transferToFailure(flowFile, String.format("%s was specified but result was empty.", MAX_ORDER.getDisplayName()));
+                    return false;
+                }
+
+                final Integer maxOrder;
+                try {
+                    maxOrder = Integer.parseInt(maxOrderStr);
+                } catch (final NumberFormatException e) {
+                    final String msg = String.format("Failed to get Maximum Order for group [%s] due to %s", groupId, e);
+                    transferToFailure(flowFile, msg, e);
+                    return false;
+                }
+
+                // Check max order.
+                if (order > maxOrder) {
+                    final String msg = String.format("Order (%d) is greater than the Maximum Order (%d) for Group [%s]", order, maxOrder, groupId);
+                    transferToFailure(flowFile, msg);
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        private boolean computeInitialOrder() {
+            // Compute initial order. Use asInteger() to check if it's a valid integer.
+            final String stateKeyOrder = STATE_TARGET_ORDER.apply(groupId);
+            try {
+                final AtomicReference<String> computedInitOrder = new AtomicReference<>();
+                groupStates.computeIfAbsent(stateKeyOrder, k -> {
+                    final String initOrderStr = initOrderProperty.evaluateAttributeExpressions(flowFile).getValue();
+                    final int initOrder = Integer.parseInt(initOrderStr);
```

Comment:
initOrder is unused, looks like its here in case the string can't be parsed (NumberFormatException)? If you remove initOrder there's still a warning so you don't have to remove it, I'm just making sure you didn't mean to use it for something other than trying to parse the string :)